### ISSUE TITLE
Incorporate quests in the player struct

### ISSUE THREAD

**Issue by gabigarc03** [uchicago-cs/chiventure#1163]
Quests are only partially implemented in chiventure - they are present in the game struct but cannot be stored in the player struct. It'd be great if the player could store a hash table of quests. See the [current implementation wiki](https://github.com/uchicago-cs/chiventure/wiki/Quests-~-Current-Implementation) for more details.

**Comment by Castel-Jaloux** [comment]
Opened communications with the action management team to determine how to implement this change.

**Comment by Castel-Jaloux** [comment]
Communications with the action management team revealed that the player struct is part of the game_state project area, which is currently unmanned. I have asked our team advisor to discuss with me possible steps to take.

**Comment by Castel-Jaloux** [comment]
I am beginning work on the player module within game_state to incorporate this feature.

**Comment by Castel-Jaloux** [comment]
quest hash table implemented in the player module. Initialization and Freeing functions finished. Operations on hash tables still to be implemented (such as add quest).

**Comment by Castel-Jaloux** [comment]
Add quest operation implemented. Working on resolving bugs observed through testing.

**Comment by Castel-Jaloux** [comment]
Progress is heavily impaired by a system of circular dependencies in the code. We are working on how to resolve this issue.

**Comment by JSerf02** [comment]
After many failed attempts, we've found a solution to the circular dependency issue, though it involves a redesign of the quest module so it can support global storage of quests in game_state and access of these global quests only through quest ids.  There is still more that needs to go into this redesign, so we have created a [separate issue](https://github.com/uchicago-cs/chiventure/issues/1311) for it. We will continue this task once the redesign is complete.

**Comment by JSerf02** [comment]
We fixed the circular dependency issue (see Issue #1311 for an explanation of the redesign) and got all of the changes to compile and pass our tests. This turned out to be a lot of changes for one pull request, however, so we will likely need to split it into multiple smaller pull requests

**Comment by philipchenaux** [comment]
Issue Score: Excellent

Comments:
Great work! But make sure that when you assign sprints in the future they should ideally not span over multiple sprints!
